crates/proto/src/domain/merkle.rs

```diff
@@ -206,3 +211,226 @@ impl From<SmtProof> for proto::primitives::SmtOpening {
         }
     }
 }
+
+// NODE INDEX
+// ------------------------------------------------------------------------------------------------
+impl From<NodeIndex> for proto::primitives::NodeIndex {
+    fn from(value: NodeIndex) -> Self {
+        proto::primitives::NodeIndex {
+            depth: value.depth() as u32,
+            value: value.value(),
+        }
+    }
+}
+impl TryFrom<proto::primitives::NodeIndex> for NodeIndex {
+    type Error = ConversionError;
+    fn try_from(index: proto::primitives::NodeIndex) -> Result<Self, Self::Error> {
+        let depth = u8::try_from(index.depth)?;
+        let value = index.value;
+        Ok(NodeIndex::new(depth, value)?)
+    }
+}
```

Comment:
Probably nothing, but we encode the depth as u32 and then cast it to u8. Should a comment be added about why is it always valid?